# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../domino/kicker.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./domino/eventfilter.cpp.orig	2006-07-22 00:03:13.000000000 +0200
+++ ./domino/eventfilter.cpp	2006-09-20 10:37:30.000000000 +0200
@@ -905,6 +905,7 @@
 		switch(ev->type()) {
 			case QEvent::Paint: {
 				KToolBar *bar = dynamic_cast<KToolBar*>(obj->parent());
+				if(!bar) return false;
 				KToolBarButton* btn = dynamic_cast<KToolBarButton*>(obj);
 				int index = bar->itemIndex(btn->id());
 				
--- ./domino/domino.cpp.orig	2006-07-22 00:03:13.000000000 +0200
+++ ./domino/domino.cpp	2006-09-20 10:38:06.000000000 +0200
@@ -4399,6 +4399,7 @@
 			
 				const bool reverse = QApplication::reverseLayout();
 				KToolBar *bar = dynamic_cast<KToolBar*>(tb->parent());
+				if(!bar) break;
 				KToolBarButton* btn = ::qt_cast<KToolBarButton*>(tb);
 				KToolBarButton* prevBtn = 0;
 				KToolBarButton* nextBtn = 0;
